### PR TITLE
Feature eden postajob

### PR DESCRIPTION
**Pull request by EdenWuyifan**
This is the PR for the following features:

- Description part for **Edit profile** & its test cases
- **Post a job** form and template (tested)
- **Post a job** view and form sketches
- Posted jobs feature on organization profile

### CODE REVIEW

File: voluncheer/profiles/models.py [hunk 1]
```diff
@@ -105,6 +105,8 @@ class Organization(models.Model):
         primary_key=True,
     )
     name = models.CharField(max_length=200)
+    website = models.CharField(max_length=200, default="")
+    description = models.TextField(help_text="Introduce yourself here.", default="")
```
Comment (joeleu) [inline voluncheer/profiles/models.py:109]: For an organization maybe it should be `Introduce your organization here.`?

File: voluncheer/profiles/test_forms.py [hunk 2]
```diff
@@ -1,7 +1,15 @@
+from datetime import datetime
+
 from django.test import TestCase
 
+from profiles.models import Organization
+from profiles.models import User
+from profiles.models import UserType
+from profiles.models import Volunteer
 from profiles.forms.volunteers import VolunteerCreationForm
```
Comment (joeleu) [inline voluncheer/profiles/test_forms.py:9]:
```suggestion
from profiles.forms.volunteers import VolunteerChangeForm
from profiles.forms.volunteers import VolunteerCreationForm
from profiles.forms.organizations import OrganizationChangeForm
from profiles.forms.organizations import OrganizationCreationForm
```

File: voluncheer/profiles/test_forms.py [hunk 3]
```diff
@@ -110,3 +118,84 @@ def test_organization_must_submit_unique_email(self):
...
+            first_name="Luke",
+            last_name="Skywalker",
+            date_of_birth="1955-09-25",
+            description="I want to come with you to Alderaan.",
+        )
+        self.user.save()
+
+    def test_validation(self):
+        """Test volunteer profile update validation"""
+        self.assertFalse(self.user.first_name == "R2D2")
```
Comment (joeleu) [inline voluncheer/profiles/test_forms.py:144]: Prefer using more accurate assertions, in this case `self.assertNotEqual`. Here and elsewhere.

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertNotEqual

**Review by joeleu — CHANGES_REQUESTED** [review]
Looks good overall, great work!

File: voluncheer/jobboard/forms/postajob.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+class PostAJobForm(forms.ModelForm):
+    """This is the form used for creating a new job for organization users"""
+
+    class Meta:
+        model = Job
+        fields = (
+            "category",
+            "title",
+            "description",
+            # "date",
```
[the remark this answers is not in the document]
Comment (EdenWuyifan) [reply voluncheer/jobboard/forms/postajob.py:19]: Leave that field for later adding a date picker

File: voluncheer/jobboard/forms/postajob.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        user = self.instance
+        organization = Organization.objects.get(pk=user)
+        if self.is_valid():
+            job = Job()
+            job.pubdate = timezone.now()
+            job.organization = organization
+            job.category = self.cleaned_data.get("category")
+            job.title = self.cleaned_data.get("title")
+            job.description = self.cleaned_data.get("description")
+            # job.date = self.cleaned_data.get("date")
```
[the remark this answers is not in the document]
Comment (EdenWuyifan) [reply voluncheer/jobboard/forms/postajob.py:36]: Save as above

File: voluncheer/jobboard/urls.py [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
 
-app_name = "jobboard"
 urlpatterns = [
-    path("", views.jobboard, name="jobboard"),
+    path("", jobboard, name="jobboard"),
     # Jobboard
-    path("select", views.select, name="select"),
+    path("select", select, name="select"),
+    # Post a Job
+    path("post_a_job", post_a_job, name="post_a_job"),
```
[the remark this answers is not in the document]
Comment (dbleek) [reply voluncheer/jobboard/urls.py:12]: On the back-end, I'm fine with job as a technical/short-hand term, but anything user-facing should definitely use opportunity. So perhaps the URL should be "opportunities/" for the job board and "opportunities/post" for the submission form. 

As for what we refer to it in the code, the only reasons I see to keep it as "job" is that it's slightly shorter to write job and it matches better with the job board (which we are are more likely say than "opportunity board"). But, in the interest of longevity, perhaps it is better to call everything opportunity to so that someone from the outside wouldn't be confused by jobs vs. opportunities.

Long story short, I agree we should change everything to opportunity/opportunities.

**Review by dbleek — APPROVED** [review]
Nothing else to add to the existing conversation except a note about jobs vs. opportunities.

File: voluncheer/profiles/views/profile.py [hunk 7]
```diff
@@ -38,10 +38,13 @@ def get_context_data(self, **kwargs):
         """Returns additional contextual information for display."""
         user = self.request.user
         if user.is_organization:
-            kwargs["organization"] = Organization.objects.get(pk=user)
+            organization_profile = Organization.objects.get(pk=user)
+            kwargs["organization"] = organization_profile
             kwargs["user_form"] = OrganizationChangeForm(
                 instance=self.request.user
             )  # noqa: E501
```
[the remark this answers is not in the document]
Comment (dbleek) [reply voluncheer/profiles/views/profile.py:45]: The line length is 84 which is over our default of 79. Should we increase the flake8 line length to 88 to match black?

File: voluncheer/profiles/test_forms.py (hunk 2 above)
Comment (dbleek) [reply voluncheer/profiles/test_forms.py:9]: I'm not sure what is being suggested here. Are you asking to switch the order of lines 9 and 10?
